Sources/Model/JSONDecorator.swift

```diff
@@ -140,317 +149,421 @@ private extension JSONDecorator {
+            let slice = createJSONSlice(expand, nil, foldedTimes, 1)
+            return [slice]
+        }
+    }
+    
+    // MARK: array
+    
+    func processArrayValue(
+        with values: [JSONValue],
+        currentSlicesCount: Int,
+        isNeed: NeedConfig,
+        foldedTimes: Int,
+        createJSONSlice: CreateJSONSlice
+    ) -> [JSONSlice] {
+        var result: [JSONSlice] = []
+        
+        let subSlicesFoldedTimes = calculateSubSlicesFoldedTimes(
+            currentState: initialState,
+            foldedTimes: foldedTimes)
+        
+        let (startExpand, startFold) = createArrayStartAttribute(
+            isNeedIndent: isNeed.indent,
+            isNeedComma: isNeed.comma)
+        
+        let slice = createJSONSlice(startExpand, startFold, foldedTimes, result.count)
+        result.append(slice)
+        
+        func _appendArrayEnd() {
+            let endExpand = createArrayEndAttribute(isNeedComma: isNeed.comma)
+            let slice = createJSONSlice(endExpand, nil, subSlicesFoldedTimes, result.count)
+            result.append(slice)
+        }
+        
+        guard !values.isEmpty else {
+            // If the array is empty, add the end flag directly.
+            _appendArrayEnd()
             return result
+        }
+        
+        incIndent()
+        
+        // Process each value
+        for (index, value) in values.enumerated() {
+            let _isNeedComma = index != (values.count - 1)
             
-        // MARK: object
-        case .object(let object):
-            let subSlicesFoldedTimes = calculateSubSlicesFoldedTimes(currentState: initialState)
-            
-            let (startExpand, startFold) = createObjectStartAttribute(
-                isNeedIndent: isNeedIndent,
-                isNeedComma: isNeedComma)
-            
-            _append(expand: startExpand, fold: startFold, foldedTimes: foldedTimes)
-            
-            func _appendObjectEnd() {
-                let endExpand = createObjectEndAttribute(isNeedComma: isNeedComma)
-                _append(expand: endExpand, fold: nil, foldedTimes: subSlicesFoldedTimes)
-            }
+            let slices = processJSONValueRecursively(
+                value,
+                currentSlicesCount: currentSlicesCount + result.count, 
+                isNeed: (indent: true, comma: _isNeedComma),
+                foldedTimes: subSlicesFoldedTimes)
             
-            guard !object.isEmpty else {
-                // If the object is empty, add the end flag directly.
-                _appendObjectEnd()
-                return result
+            result.append(contentsOf: slices)
+        }
+        
+        decIndent()
+        
+        // The end node is added only if the array is correct.
+        if case .wrong = values.last?.isRight { } else {
+            _appendArrayEnd()
+        }
+        
+        return result
+    }
+    
+    // MARK: object
+    
+    func processObjectValue(
```

Comment:
The current method body length of this method exceeds the SwiftLint limit, and the status quo is temporarily maintained for later optimization.